### PR TITLE
Fixes dropdown does not position itself at selected element when opened

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
Fixes issue: dropdown does not position itself at selected element when opened
Also fixes issue with remove all button not hiding when no option is selected.

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: #3559

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: ui/packages/form/src/DropdownOptions.svelte [hunk 1]
```diff
@@ -11,9 +11,18 @@
 	let distance_from_bottom: number;
 	let input_height: number;
 	let refElement: HTMLDivElement;
+	let listElement: HTMLUListElement;
 	let top: string | null, bottom: string | null, max_height: number;
 	$: {
 		if (showOptions && refElement) {
+			if (listElement && typeof value === "string") {
+				let el = document.querySelector(
```
Comment (space-nuko) [inline ui/packages/form/src/DropdownOptions.svelte:19]: Is this going to be compatible with #3621, when other dropdowns may have `li` elements still on the page?

**Review by aliabd — APPROVED** [review]
LGTM

File: ui/packages/form/src/DropdownOptions.svelte (hunk 1 above)
Comment (pngwn) [reply ui/packages/form/src/DropdownOptions.svelte:19]: We can address that later if we decide to add that feature.

**Review by pngwn — APPROVED** [review]
Looks good to me! Sorry for the delay on this one!

**Comment by dawoodkhan82** [comment]
Thanks for the review @pngwn @aliabd. Will merge when tests pass

**Pull request merged**